lib/grpc/stream/operators.ex

```diff
@@ -0,0 +1,172 @@
+defmodule GRPC.Stream.Operators do
+  @moduledoc """
+  Useful and internal functions for manipulating streams.
+  """
+  alias GRPC.Stream, as: GRPCStream
+
+  @type item :: any()
+
+  @type reason :: any()
+
+  @spec ask(GRPCStream.t(), pid | atom, non_neg_integer) ::
+          GRPCStream.t() | {:error, item(), reason()}
+  def ask(stream, target, timeout \\ 5000)
+
+  def ask(%GRPCStream{flow: flow} = stream, target, timeout) when is_pid(target) do
+    mapper = fn item ->
+      if Process.alive?(target) do
+        send(target, {:request, item, self()})
+
+        result =
+          receive do
+            {:response, res} -> res
+          after
+            timeout -> {:error, item, :timeout}
+          end
+
+        result
+      else
+        {:error, item, :not_alive}
+      end
+    end
+
+    %GRPCStream{stream | flow: Flow.map(flow, mapper)}
+  end
+
+  def ask(%GRPCStream{flow: flow} = stream, target, timeout) when is_atom(target) do
+    mapper = fn item ->
+      if function_exported?(target, :handle_call, 3) do
+        try do
+          case GenServer.call(target, {:request, item}, timeout) do
```

Comment:
As I suggested above, if we just `send` the message, the receive-after combo above will effectively do what GenServer.call does, and the GenServer can just implement `handle_info` to deal with this.